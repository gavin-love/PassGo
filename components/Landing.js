import React, { Component } from 'react';
import { View, Text, StyleSheet, Button, TextInput, Image } from 'react-native';
import { auth, firebase } from '../firebase';
import Geolocation from 'react-native-geolocation-service';
import CompaniesContainer from './CompaniesContainer';
// import PropTypes from 'prop-types';


class Landing extends Component {
  constructor() {
    super()
    this.state = {
      password: '',
      email: '',
      isLoading: false,
      loggedIn: false,
      pressStatus: false
    }
  }

  _onHideUnderlay() {
    this.setState({ pressStatus: false });
  }

  _onShowUnderlay() {
    this.setState({ pressStatus: true });
  }

  static navigationOptions = {
    title: 'Pass Go!'
  }

  fetchUserPoints = () => {
    return fetch('http://localhost:3000/api/v1/users')
  }

  handleSignIn = () => {
    const { email, password } = this.state;
    auth
      .doSignInWithEmailAndPassword(email, password)
      .then(user => {
        this.setState({
          loggedIn: true,
          displayName: user.user.displayName
        })
      })
      .then(() => console.log('Welcome Back!'))
      .catch(error => {
        console.log(error.message)
      })
  }

  sendPositionToBackend = (coords) => {
    return fetch('http://localhost:3000/users/:user_id/locations', {
      method: 'PUT',
      header: { 'Content-Type': 'application/json' },
      body: JSON.stringify(coords)
    })
    .then(response => response.json())
    .then(result => console.log(result))
  }

  componentDidMount() {
    firebase.auth
      .onAuthStateChanged(user => {
        if (user) {
          this.setState({
            loggedIn: true,
            userName: user.displayName
          })
        }
      })
    // Geolocation.watchPosition(
    //   (position) => {
    //     this.sendPositionToBackend(position);
    //   },
    //   (error) => {
    //     console.log(error.message)
    //   },
    //   { enableHighAccuracy: true, distanceFilter: 10, maximumAge: 0 }
    // )
  }

  render() {
    const { navigate } = this.props.navigation;
    if (!this.state.loggedIn) {
      return (
        <View style={styles.page}>
          <Image source={require('./passgo1.png')} style={styles.logo}/>
          <View>
            <Text style={styles.inputTitle}>Email</Text>
            <TextInput style={styles.textInput} value={this.state.email} onChangeText={(email) => this.setState({ email })} />
            <Text style={styles.inputTitle}>Password</Text>
            <TextInput style={styles.textInput} value={this.state.password} onChangeText={(password) => this.setState({ password })} />
            <View style={styles.submit}>
              <Button title='submit' color="#f3f3f3" onPress={this.handleSignIn} />
            </View>
            <View style={styles.submit}>
              <Button title="sign up" color="#f3f3f3" onPress={() => {
                navigate('SignUp');
              }} />
            </View>
          </View>
        </View>
      )
    } else {
      return (
        <View style={styles.page}>
          <Text>{`Welcome ${this.state.userName}!`}</Text>
          <View>
            <CompaniesContainer companies={[{ name: 'SliceWorks', points: 40 }]}/>
          </View>
          <View style={styles.signOut}>
            <Button title="Sign Out" onPress={() => {
              auth.doSignOut()
                .then(() => this.setState({ loggedIn: false }))
                .catch(error => console.log(error.message))
            }} />
          </View>
        </View>
      )
    }
  }
}

const styles = StyleSheet.create({
  page: {
    alignItems: 'center',
    backgroundColor: '#3383bb',
    height: 600
  },
  inputTitle: {
    color: '#f3f3f3'
  },
  textInput: {
    height: 40,
    borderColor: '#24445b',
    borderWidth: 1,
    width: 200
  },
  submit: {
    borderColor: '#24445b',
    borderWidth: 1,
    height: 30,
    borderRadius: 5,
    margin: 5,
    lineHeight: 13
  },
  signOut: {
    borderColor: '#24445b',
    borderWidth: 1,
    width: 100
  },
  logo: {
    width: 300,
    height: 300
  }
})

export default Landing;