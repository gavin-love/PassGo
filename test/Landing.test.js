import React, { Component } from 'react';
import Landing from '../components/Landing';
import TestRenderer from 'react-test-renderer';

it('renders correctly if user is not logged in', () => {
  const navigation = {
    navigate: jest.fn()
  }
  const wrapper = TestRenderer.create(<Landing navigation={navigation}/>);
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly if user is logged in', () => {
  const navigation = {
    navigate: jest.fn()
  }
  const wrapper = TestRenderer.create(<Landing navigation={navigation} />);
  wrapper.getInstance().setState({ loggedIn: true });
  expect(wrapper).toMatchSnapshot();
});