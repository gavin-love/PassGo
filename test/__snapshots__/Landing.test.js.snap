// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3383bb",
      "height": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../components/passgo1.png",
      }
    }
    style={
      Object {
        "height": 300,
        "width": 300,
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "color": "#f3f3f3",
          "fontSize": 15,
          "marginBottom": 5,
        }
      }
    >
      Email
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "#24445b",
          "borderWidth": 1,
          "height": 40,
          "shadowColor": "#f3f3f3",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "color": "#f3f3f3",
          "fontSize": 15,
          "marginBottom": 5,
          "marginTop": 15,
        }
      }
    >
      Password
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "#24445b",
          "borderWidth": 1,
          "height": 40,
          "shadowColor": "#f3f3f3",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "#24445b",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "marginTop": 15,
          "shadowColor": "#f3f3f3",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 200,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            disabled={false}
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#f3f3f3",
                },
              ]
            }
          >
            submit
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "#24445b",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "lineHeight": 13,
          "margin": 5,
          "marginTop": 60,
          "shadowColor": "#f3f3f3",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 80,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            disabled={false}
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#f3f3f3",
                },
              ]
            }
          >
            sign up
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly if user is logged in 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3383bb",
      "height": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#f3f3f3",
        "fontSize": 25,
      }
    }
  >
    Welcome !
  </Text>
  <View>
    <View
      style={
        Object {
          "marginTop": 50,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#f3f3f3",
            "fontSize": 20,
            "marginBottom": 15,
          }
        }
      >
        Your Companies:
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderColor": "#24445b",
        "borderWidth": 1,
        "marginTop": 10,
        "shadowColor": "#f3f3f3",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 2,
        "width": 100,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#f3f3f3",
              },
            ]
          }
        >
          Sign Out
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly if user is not logged in 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3383bb",
      "height": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../components/passgo1.png",
      }
    }
    style={
      Object {
        "height": 300,
        "width": 300,
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "color": "#f3f3f3",
          "fontSize": 15,
          "marginBottom": 5,
        }
      }
    >
      Email
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "#24445b",
          "borderWidth": 1,
          "height": 40,
          "shadowColor": "#f3f3f3",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "color": "#f3f3f3",
          "fontSize": 15,
          "marginBottom": 5,
          "marginTop": 15,
        }
      }
    >
      Password
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "#24445b",
          "borderWidth": 1,
          "height": 40,
          "shadowColor": "#f3f3f3",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "#24445b",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "marginTop": 15,
          "shadowColor": "#f3f3f3",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 200,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            disabled={false}
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#f3f3f3",
                },
              ]
            }
          >
            submit
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "#24445b",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "lineHeight": 13,
          "margin": 5,
          "marginTop": 60,
          "shadowColor": "#f3f3f3",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 80,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            disabled={false}
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#f3f3f3",
                },
              ]
            }
          >
            sign up
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
